### ISSUE TITLE
Temporal services take a long time to start with degraded Cassandra cluster

### ISSUE THREAD

**Issue by scoplin** [temporalio/temporal#3638]
## Expected Behavior
In a 3-node Cassandra cluster setup, I would expect that service availability would be maintained even if one of those hosts was down.  In a containerized Temporal environment, availability includes the ability to restart pods, as it may be needed for scaling or fault recovery.  This means that a degraded (but technically available) cluster should not block startup for Temporal pods.

## Actual Behavior
When one of the hosts of our Cassandra cluster is down, newly booted temporal pods of varying types (history, frontend, etc.) take nearly an hour to begin to service user requests successfully.  During the startup sequence, they report errors such as:

```
2022/11/21 20:53:59 gocql: unable to dial control conn xx.xx.xx.xx:9042: dial tcp xx.xx.xx.xx:9042: i/o timeout
```
(IP is the address of the Cassandra host that is down.)

## Steps to Reproduce the Problem

  1. Configure Cassandra as a Temporal persistence store.
  1. Shut down one of the Cassandra nodes.
  1. Restart a temporal service (history, frontend, etc.)

## Specifications

  - Version: 1.18.0
  - Platform: Kubernetes/Cassandra self-hosted

**Comment by scoplin** [comment]
I think this might be related to https://github.com/gocql/gocql/issues/1365

I’m wondering if wrapping the [token aware host selection policy](https://github.com/temporalio/temporal/blob/master/common/persistence/nosql/nosqlplugin/cassandra/gocql/client.go#L161) with a [single host ready policy](https://github.com/gocql/gocql/blob/6132f789ad22c1d19a66c58ac789402d5f1f9be5/policies.go#L960) might not fix the issue?

**Comment by yiminc** [comment]
Gocql 1.4.0 should have improved this. It is included in temporal server v1.21.5.
